src/domain/core/src/repos/metadata_block_repository.rs

```diff
@@ -0,0 +1,245 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use async_trait::async_trait;
+use bytes::Bytes;
+use chrono::{DateTime, Utc};
+use internal_error::InternalError;
+use opendatafabric::{MetadataBlock, Multihash};
+use thiserror::Error;
+use url::Url;
+
+use crate::{
+    AccessError,
+    BlockNotFoundError,
+    ContainsError,
+    ExternalTransferOpts,
+    GetBlockError,
+    GetError,
+    GetExternalUrlError,
+    GetExternalUrlResult,
+    HashMismatchError,
+    InsertError,
+    InsertOpts,
+    InsertResult,
+    ObjectNotFoundError,
+};
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait]
+pub trait MetadataBlockRepository: Send + Sync {
+    async fn contains_block(&self, hash: &Multihash) -> Result<bool, ContainsBlockError>;
+
+    async fn get_block(&self, hash: &Multihash) -> Result<MetadataBlock, GetBlockError>;
+
+    async fn get_block_data(&self, hash: &Multihash) -> Result<Bytes, GetBlockDataError>;
+
+    async fn get_block_size(&self, hash: &Multihash) -> Result<u64, GetBlockDataError>;
+
+    async fn insert_block_data<'a>(
+        &'a self,
+        block_data: &'a [u8],
+        options: InsertOpts<'a>,
+    ) -> Result<InsertBlockResult, InsertBlockError>;
+
+    async fn get_block_external_download_url(
+        &self,
+        hash: &Multihash,
+        opts: ExternalTransferOpts,
+    ) -> Result<GetBlockExternalUrlResult, GetBlockExternalUrlError>;
+
+    async fn get_block_external_upload_url(
+        &self,
+        hash: &Multihash,
+        opts: ExternalTransferOpts,
+    ) -> Result<GetBlockExternalUrlResult, GetBlockExternalUrlError>;
```

Comment:
Not sure about these methods - will bring up on Slack

Reply:
After [the hotfix](https://github.com/kamu-data/kamu-cli/pull/496), we don't need this structure -- we can just cache blocks, not bytes

Thus, methods working with bytes (insert, delete, get size) are no longer needed -- only block-level methods are left in `MetadataBlockRepository`